### PR TITLE
Agent start location randomization #295

### LINKED ISSUE #295: Optionally add a slight bit of randomization to agent start location
David mentioned that it might be good to have the start location of agents randomized a bit (less than half a meter or so in each direction) on reset to add a bit of entropy.

It would be good if this were configurable per-scenario in the config file, ie "start location randomization"

### PR DESCRIPTION
**Pull request by daniekpo**
Added optional randomization to agents' start location. This fixes #295

### CODE REVIEW

**Comment by jaydenmilne** [comment]
Could you write a few tests to validate this? Shouldn't be too difficult

**Comment by jaydenmilne** [comment]
Sorry I thought I had filled out that issue more. It would be better I think if this were per agent instead of scenario wide - i.e. 
```python
{
   "agent_name": "uav0",
   "agent_type": "{agent types}",
   "sensors": [
      "array of sensor objects"
   ],
   "control_scheme": "{control scheme type}",
   "location": [1.0, 2.0, 3.0],
   "rotation": [1.0, 2.0, 3.0],
   "location_randomization": [dx, dy, dz],
   "rotation_randomization": [dpitch, droll, dyaw]
}
```

`location_randomization` would be optional

**Comment by daniekpo** [comment]
Will do!

File: tests/agents/test_randomization.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    with HolodeckEnvironment(scenario=conf, binary_path=bin_path, show_viewport=False, uuid=str(uuid.uuid4())) as env:
+        prev_location = conf["agents"][0]["location"]
+        default_start_location = conf["agents"][0]["location"]
+        variance = conf["agents"][0]["location_randomization"]
+
+        num_resets = 5
+        for _ in range(num_resets):
+            cur_location = env.tick()["LocationSensor"]
+
+            assert is_different_3d_vector(cur_location, prev_location)
```
Comment (jaydenmilne) [inline tests/agents/test_randomization.py]: Tests look great!. Just so you know, you can put a string after an assert to provide a more helpful error message.

```python
assert is_different_3d_vector(cur_location, prev_location), \
    "The location vector didn't change! current = {} previous = {}".format(cur_location, prev_location)
```

I'm not sure I've been super religious about this in the test suite though

Comment (jaydenmilne) [reply tests/agents/test_randomization.py]: (I'm not saying have to do this, just an FYI)

**Pull request merged**